### PR TITLE
SASS-1088: Move some logic from controllers into service layer

### PR DESCRIPTION
**Pull request by jhkldev**
### Description
Refactor logic from controllers into the service layer.

[SASS-1088](https://jira.tools.tax.service.gov.uk/browse/SASS-1088)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [X]  Have you run the tests?
- [X]  Have you run the journey tests? (where applicable)
- [X]  Have you addressed warnings where appropriate?
- [X]  Have you rebased against the current version of main?
- [X]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [X]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by robert-webb-hmrc** [comment]
test this please

File: app/models/interest/TaxedModel.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+package models.interest
+
+import common.InterestTaxTypes.TAXED
+
+case class TaxedModel(accountName: String, accountAmount: BigDecimal)
+
+object TaxedModel {
```
Comment (robert-webb-hmrc) [inline app/models/interest/TaxedModel.scala:23]: I'm not sure this should be called TaxedModel if it's being used to map both types

File: test/models/interest/TaxedModelSpec.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.interest
+
+import common.InterestTaxTypes
+import utils.UnitTest
+
+class TaxedModelSpec extends UnitTest {
```
Comment (robert-webb-hmrc) [inline test/models/interest/TaxedModelSpec.scala:22]: See above comment

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Please see comments - could we also cover the following if possible?
Screenshot 2022-02-01 at 09 16 13
Screenshot 2022-02-01 at 09 11 46

File: test/models/interest/AccountAmountModelSpec.scala [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    InterestAccountModel(
+      None,
+      "Account 1",
+      taxedAmount = Some(100.01),
+      uniqueSessionId = Some("3")
+    )
+  )
+
+  "apply" should {
+    "return instance of TaxedModel containing name and tax amount if cya data, valid unique session id and tax type 'TAXED' is specified" in {
```
Comment (robert-webb-hmrc) [inline test/models/interest/AccountAmountModelSpec.scala:45]: Could we change these test names to reflect the file name please?

File: test/models/interest/AccountAmountModelSpec.scala [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    "return instance of TaxedModel containing name and untaxed amount if cya data, valid unique session id and tax type 'UNTAXED' is specified" in {
+      val cyaModel = Some(InterestCYAModel(
+        untaxedUkInterest = Some(false),
+        taxedUkInterest = Some(true),
+        accounts = accounts
+      ))
+
+      val taxedModel = AccountAmountModel.apply(cyaModel, "1", InterestTaxTypes.UNTAXED)
```
Comment (robert-webb-hmrc) [inline test/models/interest/AccountAmountModelSpec.scala:64]: (and val names too)

File: test/models/interest/AccountAmountModelSpec.scala (hunk 3 above)
Comment (jhkldev) [reply test/models/interest/AccountAmountModelSpec.scala]: Done 👍

File: test/models/interest/AccountAmountModelSpec.scala (hunk 4 above)
Comment (jhkldev) [reply test/models/interest/AccountAmountModelSpec.scala]: Done 👍

**Review by robert-webb-hmrc — APPROVED** [review]

**Pull request merged**